### PR TITLE
Bug 1815015 - Set the engine session for background tabs inactive

### PR DESCRIPTION
**Pull request by kycn**
When a tab is created in the background via "Open Link in New Tab", its engine session should be inactive.

Note: This is a draft, no unit tests are written at the moment. This api change is the last-resort in case the problem is not fixable in the GeckoView APIs internally. 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1815015

### CODE REVIEW

**Comment by kycn** [comment]
@jonalmeida i'm not very happy with the amount of parameter passing across the layers here, if you have a better solution that has aligns with the apis and the architecture more, i would be glad to discuss that 😄

**Comment by jonalmeida** [comment]
> @jonalmeida i'm not very happy with the amount of parameter passing across the layers here, if you have a better solution that has aligns with the apis and the architecture more, i would be glad to discuss that 😄

Is this a direction you are looking to pursue now? Have we reached the "last resort" phase? 🙃

**Comment by kycn** [comment]
> > @jonalmeida i'm not very happy with the amount of parameter passing across the layers here, if you have a better solution that has aligns with the apis and the architecture more, i would be glad to discuss that 😄
> 
> Is this a direction you are looking to pursue now? Have we reached the "last resort" phase? 🙃

Unfortunately, it looks like that. The failures in GV do not explain themselves quite well... 
I timescoped my investigations of the failures until next Monday EOD. In case I cannot resolve the issues there, either because of the tech limitations or the time constraints, I will be taking this PR ready for review 😄 
So on Monday EOD, this PR's future will be decided.

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 1]
```diff
@@ -1298,6 +1299,7 @@ class GeckoEngineSession(
         geckoSession.historyDelegate = createHistoryDelegate()
         geckoSession.mediaSessionDelegate = GeckoMediaSessionDelegate(this)
         geckoSession.scrollDelegate = createScrollDelegate()
+        geckoSession.setActive(!isBackground)
```
[the remark this answers is not in the document]
Comment (csadilek) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:1302]: (I should add that I don't oppose this solution, just want us to understand and document why we couldn't do it in GV.)

[the remark this answers is not in the document]
Comment (kycn) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:1302]: Another option I'm currently testing:
- adding `geckoSession.setActive(false)` in `GeckoEngineSession.createGeckoSession()` method. This is where we create the GeckoSession in AC APIs. Both foreground and background tabs/sessions call this method, but the background tabs do not call `GeckoView.setSession()` method which sets the activeness to true. That's why the session activeness remains false for background tabs, while it is activated in foreground tabs.

**Comment by kycn** [comment]
Closing this PR as the solution is addressed in GeckoView codebase. See this patch: https://phabricator.services.mozilla.com/D176421

**Pull request closed without merging**